### PR TITLE
Mask boundary nodes in spatial loss maps during test_step

### LINKED ISSUE #569: [Bug] spatial_loss_maps in test_step computed without interior_mask, inconsistent with all other loss calls
**Description**

In `ar_model.py`, `test_step` computes `spatial_loss_maps` at L452-454 without passing `mask=self.interior_mask_bool`:

```python
spatial_loss = self.loss(
    prediction, target, pred_std, average_grid=False
)
```

Every other call to `self.loss` in the module passes `mask=self.interior_mask_bool`:

- `training_step` (L311)
- `validation_step` (L354)
- `test_step` main loss (L408)

The per-variable metrics (`mse`, `mae`) computed just above (L435-441) also pass the mask. This means the spatial loss maps include boundary grid nodes that are excluded everywhere else.

**Impact**

- The saved `mean_spatial_loss.pt` contains loss values at boundary nodes that the model never optimises for, which can mislead downstream error analysis.
- `plot_spatial_error` picks `vmin`/`vmax` from the full tensor, so boundary-node loss values can skew the colour scale.
- For LAM domains with boundary forcing the inconsistency is most pronounced; for global domains (no boundary) it has no practical effect.

**Minimal example**

```bash
python -m neural_lam.train_model --eval test --config_path <config> --load <ckpt>
# spatial_loss_maps saved to <logger_dir>/spatial_loss_maps/ include boundary values
```

**Proposed fix**

After computing spatial loss over the full grid (to preserve the `num_grid_nodes` shape needed by plotting), mask boundary nodes to NaN:

```python
spatial_loss = self.loss(
    prediction, target, pred_std, average_grid=False
)
spatial_loss[..., ~self.interior_mask_bool] = float("nan")
```

And use `torch.nanmean` instead of `torch.mean` when averaging the maps in `on_test_epoch_end`.

### PR DESCRIPTION
**Pull request by RajdeepKushwaha5**
## Describe your changes

The `spatial_loss` call in `test_step` did not pass `mask=self.interior_mask_bool`, unlike every other `self.loss()` call in `training_step`, `validation_step`, and `test_step` itself. This caused boundary grid nodes to be included in the spatial loss maps and the saved `mean_spatial_loss.pt`, inconsistent with the loss the model optimises.

Set boundary node values to NaN after computing the full-grid loss (to keep the `num_grid_nodes` shape needed by `plot_spatial_error`) and switch to `torch.nanmean` when averaging over samples in `on_test_epoch_end`.

## Issue Link

closes #569

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (Addition or improvements to documentation)

## Checklist before requesting a review

- [x] My branch is up-to-date with the target branch
- [x] I have performed a self-review of my code
- [x] For any new/modified functions/classes I have added docstrings that clearly describe its purpose, expected inputs and returned values
- [x] I have placed in-line comments to clarify the intent of any hard-to-understand passages of my code
- [ ] I have updated the [README](README.MD) to cover introduced code changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have given the PR a name that clearly describes the change, written in imperative form
- [ ] I have requested a reviewer and an assignee

## Checklist for reviewers

- [ ] the code is readable
- [ ] the code is well tested
- [ ] the code is documented (including return types and parameters)
- [ ] the code is easy to maintain

## Author checklist after completed review

- [x] I have added a line to the CHANGELOG describing this change, in a section reflecting type of change

## Checklist for assignee

- [ ] PR is up to date with the base branch

### CODE REVIEW

File: neural_lam/models/ar_model.py [hunk 1]
```diff
@@ -447,6 +447,8 @@ def test_step(self, batch, batch_idx):
         spatial_loss = self.loss(
             prediction, target, pred_std, average_grid=False
         )  # (B, pred_steps, num_grid_nodes)
+        # Exclude boundary nodes, consistent with training/validation loss
```
Comment (sadamov) [inline neural_lam/models/ar_model.py:450]: File is gone after the #208 refactor, this block should land in `models/module.py:442` right after `spatial_loss = self.loss(...)`:

```python
        # Exclude boundary nodes, consistent with the loss used in
        # training/validation. NaN-mask in place so the tensor keeps its
        # (B, pred_steps, num_grid_nodes) shape for downstream plotting,
        # and is reduced with `torch.nanmean` in `on_test_epoch_end`.
        spatial_loss[..., ~self.interior_mask_bool] = float("nan")
```

File: neural_lam/models/ar_model.py [hunk 2]
```diff
@@ -753,7 +755,7 @@ def on_test_epoch_end(self):
             torch.cat(self.spatial_loss_maps, dim=0)
         )  # (N_test, N_log, num_grid_nodes)
         if self.trainer.is_global_zero:
-            mean_spatial_loss = torch.mean(
+            mean_spatial_loss = torch.nanmean(
```
Comment (sadamov) [inline neural_lam/models/ar_model.py:758]: Same, apply at `models/module.py:722`:

```python
            mean_spatial_loss = torch.nanmean(spatial_loss_tensor, dim=0)
```

**Review by sadamov — COMMENTED** [review]
`ar_model.py` was removed by #208, needs a rebase onto current main. Fix moves to `models/module.py`. Also I'd drop the standalone test, see inline.
If you are willing to bring this over the finish line over two months later, you can clear the conflicts and I'll merge.

File: tests/test_training.py [hunk 3]
```diff
@@ -189,3 +189,50 @@ def all_gather(self, tensor, sync_grads=False):
         "all_gather_cat produced incorrectly ordered/combined values "
         "on multi-device simulation"
     )
+
```
Comment (sadamov) [inline tests/test_training.py:192]: I'd drop this one. It only verifies that `torch.nanmean` ignores NaN, which is testing PyTorch behavior rather than this integration. Could you replace it with an integration test that builds a `ForecasterModule` with a known `interior_mask`, runs `trainer.test()` end-to-end, and asserts boundary nodes are NaN in the saved `mean_spatial_loss.pt`? That would also be the first real coverage for `test_step` / `on_test_epoch_end` in this repo.

**Comment by sadamov** [comment]
@RajdeepKushwaha5 thanks for spotting and fixing this, sorry it sat for so long. Since the rebase never happened, @NoiceHax picked up your fix and rebased it onto `models/module.py` in #720, together with the integration test I asked for above. Your change is in there unmodified and you are credited in the PR body, I will make sure the CHANGELOG line credits you too.
